### PR TITLE
Disable build targets associated with use_partition_alloc and enable_pkeys

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/382115610
b/371241293

For linux builds:
Remove partition_alloc target in //base
Remove header files causing errors during gn check
Add memory_starboard.cc to replace memory_linux.cc

There's a chunk of files in the targets which are causing compiler errors with llvm libc++, musl libc - these files weren't being compiled in 25lts. Removing these temporarily until the llvm libc++, musl libc is setup - essential to setting up the evergreen toolchain.

### CODE REVIEW

**Comment by niranjanyardi** [comment]
I made a small change in this one , more might be needed and I'll submit them one by one if needed

File: base/BUILD.gn [hunk 1]
```diff
@@ -1725,9 +1725,15 @@ component("base") {
     # partition_alloc code if use_partition_alloc = false because no code uses
     # partition_alloc.
     public_deps += [
-      "allocator/partition_allocator:partition_alloc",
       "allocator/partition_allocator:raw_ptr",
     ]
+
+    # TODO(cobalt, b/382115610): Re-enable use_partition_alloc
+    if (!is_cobalt) {
```
Comment (andrewsavage1) [inline base/BUILD.gn:1732]: Use `if (is_cobalt) { public_deps -= ... }` instead

File: base/allocator/partition_allocator/partition_alloc.gni [hunk 2]
```diff
@@ -47,7 +47,9 @@ declare_args() {
   #     and doesn't wish to incur the library size increase (crbug.com/674570).
   # 2.  On NaCl (through this declaration), where PartitionAlloc doesn't
   #     build at all.
-  use_partition_alloc = !is_nacl
+
+# TODO(cobalt, b/382115610): Re-enable use_partition_alloc
```
Comment (andrewsavage1) [inline base/allocator/partition_allocator/partition_alloc.gni:51]: This should probably be done in args.gn files

File: base/allocator/partition_allocator/partition_alloc.gni [hunk 3]
```diff
@@ -305,7 +307,8 @@ assert(!use_asan_backup_ref_ptr || use_hookable_raw_ptr,
        "AsanBackupRefPtr requires RawPtrHookableImpl")
 
 declare_args() {
-  enable_pkeys = is_linux && target_cpu == "x64"
+  # TODO(cobalt, b/382115610): Re-enable enable_pkeys
+  enable_pkeys = is_linux && target_cpu == "x64" && !is_cobalt
```
Comment (andrewsavage1) [inline base/allocator/partition_allocator/partition_alloc.gni:311]: same here

**Review by andrewsavage1 — CHANGES_REQUESTED** [review]

File: base/BUILD.gn (hunk 1 above)
Comment (niranjanyardi) [reply base/BUILD.gn:1732]: ahh the positive conditional, gotcha

File: base/allocator/partition_allocator/partition_alloc.gni (hunk 2 above)
Comment (niranjanyardi) [reply base/allocator/partition_allocator/partition_alloc.gni:51]: duh, i got a bit lost looking at all the base changes and didnt realize this :) . thanks for pointing this out

File: base/BUILD.gn (hunk 1 above)
Comment (niranjanyardi) [reply base/BUILD.gn]: done , i try to incorporate this by default in future cl's

**Comment by niranjanyardi** [comment]
> Should #4524 be closed?

done, closed

File: cobalt/build/configs/linux-x64x11/args.gn [hunk 4]
```diff
@@ -8,6 +8,11 @@ use_ozone = true
 # Disable Vulkan until we formally decide to support it.
 enable_vulkan = false
 
+
 # Overriding flag from //media/media_options.gni.
 # Cobalt doesn't use //third_party/ffmpeg.
 media_use_ffmpeg = false
+
+# TODO(b/382115610): Re-enable use_partition_alloc, enable_pkeys
+use_partition_alloc = !is_nacl
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply cobalt/build/configs/linux-x64x11/args.gn]: done.Local change never got pushed to remote

**Comment by niranjanyardi** [comment]
@andrewsavage1 please review commits on Dec5 : https://github.com/youtube/cobalt/pull/4526/commits

**Comment by niranjanyardi** [comment]
@andrewsavage1 ptal when free

File: base/BUILD.gn [hunk 5]
```diff
@@ -4016,6 +4024,69 @@ test("base_unittests") {
       "tracing/perfetto_task_runner_unittest.cc",
     ]
 
+    # TODO(cobalt, b/382115610): Re-enable use_partition_alloc
```
Comment (andrewsavage1) [inline base/BUILD.gn:4027]: This looks like more than just partition_alloc stuff, or do all of these files have linking issues with it?

File: base/metrics/field_trial.cc [hunk 6]
```diff
@@ -192,7 +192,10 @@ void OnOutOfMemory(size_t size) {
 #if BUILDFLAG(IS_NACL)
   NOTREACHED();
 #else
+// TODO(b/382115610): Re-enable use_partition_alloc
+#if !BUILDFLAG(IS_STARBOARD)
```
Comment (andrewsavage1) [inline base/metrics/field_trial.cc:196]: Is this what C25 does?

File: base/BUILD.gn (hunk 5 above)
Comment (niranjanyardi) [reply base/BUILD.gn:4027]: a chunk of them do. after adding them one by one for 4-5 files, i just copied over the entire section from 25lts - do iterate faster. We can address these whenever we re-enable use_partition_alloc .. is that okay ?

File: base/metrics/field_trial.cc (hunk 6 above)
Comment (niranjanyardi) [reply base/metrics/field_trial.cc:196]: #if BUILDFLAG(IS_NACL) || defined(STARBOARD)
  NOTREACHED();
#else


ahh C25 does this ^ . I'll copy over the same idiom

Comment (niranjanyardi) [reply base/metrics/field_trial.cc]: done,copied over from 25lts

File: base/process/memory_linux.cc [hunk 7]
```diff
@@ -111,7 +111,10 @@ bool UncheckedMalloc(size_t size, void** result) {
 #elif defined(MEMORY_TOOL_REPLACES_ALLOCATOR) || !defined(LIBC_GLIBC)
   *result = malloc(size);
 #elif defined(LIBC_GLIBC)
+// TODO(cobalt, b/382115610): Re-enable use_partition_alloc
+#if !BUILDFLAG(IS_STARBOARD)
```
Comment (andrewsavage1) [inline base/process/memory_linux.cc:115]: This isn't what we want to do—if we use this file in c25, do what we do there, but I assume we use memory_starboard.cc instead. If that's the case, then we likely want to ensure either `MEMORY_TOOL_REPLACES_ALLOCATOR` is defined or `LIBC_GLIBC` isn't—I'm not sure which is more appropriate.

Comment (niranjanyardi) [reply base/process/memory_linux.cc:115]: thanks, ill try adding memory_starboard.cc

Comment (niranjanyardi) [reply base/process/memory_linux.cc]: done, added

File: .github/config/linux.json [hunk 8]
```diff
@@ -6,14 +6,6 @@
     ],
     "targets": [
  	"base_unittests",
-        "sql_unittests",
```
Comment (niranjanyardi) [inline .github/config/linux.json:9]: All these targets fail with the existing changes. Enabling use_partition_alloc and restoring the dependency of //base on partition_alloc:partition_alloc is not trivial .  I can work on this after setting up the evergreen toolchain / someone else can work on this from the base migration team . wdyt @andrewsavage1 , @kaidokert

**Comment by niranjanyardi** [comment]
I'll clsoe this PR for now as decided in today's meeting to enable use_partition_alloc

**Pull request closed without merging**